src/Microsoft.VisualStudio.Threading/TplExtensions.cs

```diff
@@ -761,6 +883,53 @@ public void GetResult()
             }
         }
 
+        /// <summary>
+        /// Provides access to a <see cref="System.Threading.CancellationToken"/> that combines multiple other tokens,
+        /// and allows convenient disposal of any applicable <see cref="CancellationTokenSource"/>.
+        /// </summary>
+        public struct CombinedCancellationToken : IDisposable
```

Comment:
:bulb: `readonly struct`

:bulb: Consider making this a top-level type. While it's vital users dispose of it properly, it seems reasonable that it would be stored for disposing at a later point.

Reply:
> `readonly struct`

Done.

> Consider making this a top-level type. 

I think I see your point. It's not particularly the use case I'm going for though. Everywhere I've seen where multiple `CancellationToken` instances must be combined it's within a method and therefore only ever required as a local where `var` would suffice. 
The cost of another top-level type is of course that it adds to completion lists and it's not a top-level scenario to engage with this type directly.
So I'm torn.

As to `readonly struct`